src/shared_gui_components/OSGridController.hpp

```diff
@@ -110,9 +110,9 @@ class DataSource
   explicit DataSource(const std::function<std::vector<ItemType>(InputType)>& t_sourceFunc, bool t_wantsPlaceholder = false,
                       const QSharedPointer<BaseConcept>& t_dropZoneConcept = QSharedPointer<BaseConcept>())
     : m_wantsPlaceholder(t_wantsPlaceholder), m_dropZoneConcept(t_dropZoneConcept) {
-    typedef decltype(t_sourceFunc) IncommingFuncType;
-    typedef typename std::remove_reference<typename std::remove_cv<IncommingFuncType>::type>::type FunctionType;
-    typedef typename std::remove_reference<typename std::remove_cv<typename FunctionType::argument_type>::type>::type ParamType;
+
+    // Remove const and reference qualifiers: passing a function that takes `const ModelObject&` => `ModelObject` so we can use that for casting
+    using ParamType = typename std::remove_reference<typename std::remove_cv<InputType>::type>::type;
```

Comment:
This is pretty wild, I didn't know this existed, but I guess it's been there since C++11.

Reply:
The fix is simple and logical, but I had to test it out to make sure I wasn't doing something stupid (it felt off to see the complication given than InputType is passed as a template param to begin with...)

https://gcc.godbolt.org/z/6WdErGzbe